src/com/amazon/kinesis/streaming/agent/tailing/SourceFile.java

```diff
@@ -114,6 +115,13 @@ public String toString() {
     private void validateDirectory(Path dir) {
         Preconditions.checkArgument(dir != null, "Directory component is empty!");
         // TODO: validate that the directory component has no glob characters
+
+        //check the permissions of every level in the directory path
+        for(Path level : dir.toAbsolutePath()){
+            if(!Files.isReadable(level)){
+                Logging.getLogger(getClass()).warn("Permission Denied: Agent user unable to read " + dir.toString());
```

Comment:
Warnings won't prevent the agent from stopping completely so if there are permissions issues on multiple directories, every directory with an issue will appear in the logs rather than an error on the first restricted directory